test/basic.js

```diff
@@ -414,6 +414,38 @@ test('restore QoS 0 subscriptions not clean', function (t) {
   })
 })
 
+test('do not restore QoS 0 subscriptions when clean', function (t) {
+  var broker = aedes()
+  var publisher
+  var subscriber = connect(setup(broker), { clean: true, clientId: 'abcde' })
+
+  subscribe(t, subscriber, 'hello', 0, function () {
+    subscriber.inStream.end()
```

Comment:
you should check after this, not after reconnection.

Reply:
thanks! corrected that, now it fails on a master without a patch from @sundryp